lib/cl/cmdqueue.jl

```diff
@@ -81,12 +85,23 @@ function Base.getproperty(q::CmdQueue, s::Symbol)
         return Int(ref_count[])
     elseif s == :properties
         props = Ref{cl_command_queue_properties}()
-        clGetCommandQueueInfo(q, CL_QUEUE_PROPERTIES, sizeof(cl_command_queue_properties),
-                              props, C_NULL)
+        clGetCommandQueueInfo(
+            q, CL_QUEUE_PROPERTIES, sizeof(cl_command_queue_properties),
+            props, C_NULL
+        )
         return props[]
     else
         return getfield(q, s)
     end
 end
 
+function global_queue(ctx::Context, dev::Device)
+    # NOTE: dev purposefully does not default to context() or device() to stress that
+    #       objects should track ownership, and not rely on implicit global state.
+    return get!(task_local_storage(), (:CLCommandQueue, ctx, dev)) do
+        CmdQueue(ctx, dev)
+    end
+end
```

Comment:
What's wrong with `cl.queue` from `state.jl`?

Reply:
This is a testing artifact, we don't use or need this, will remove this.